src/Symfony/Component/Security/Core/Authentication/Authenticator/HttpBasicAuthenticator.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authentication\Authenticator;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Encoder\EncoderFactoryInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Guard\AuthenticatorInterface;
+
+/**
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ */
+class HttpBasicAuthenticator implements AuthenticatorInterface
+{
+    use UserProviderTrait, UsernamePasswordTrait {
+        UserProviderTrait::getUser as getUserTrait;
+    }
+
+    protected $realmName;
+    private $userProvider;
+    private $encoderFactory;
+    private $logger;
+
+    public function __construct(string $realmName, UserProviderInterface $userProvider, EncoderFactoryInterface $encoderFactory, ?LoggerInterface $logger = null)
+    {
+        $this->realmName = $realmName;
+        $this->userProvider = $userProvider;
+        $this->encoderFactory = $encoderFactory;
+        $this->logger = $logger;
+    }
+
+    public function start(Request $request, AuthenticationException $authException = null)
+    {
+        $response = new Response();
+        $response->headers->set('WWW-Authenticate', sprintf('Basic realm="%s"', $this->realmName));
+        $response->setStatusCode(401);
+
+        return $response;
+    }
+
+    public function supports(Request $request): bool
+    {
+        return $request->headers->has('PHP_AUTH_USER');
+    }
+
+    public function getUser($credentials, UserProviderInterface $userProvider): ?UserInterface
+    {
+        return $this->getUserTrait($credentials, $this->userProvider);
+    }
+
+    public function getCredentials(Request $request)
+    {
+        return [
+            'username' => $request->headers->get('PHP_AUTH_USER'),
+            'password' => $request->headers->get('PHP_AUTH_PW', ''),
+        ];
+    }
+
+    public function onAuthenticationSuccess(Request $request, TokenInterface $token, $providerKey): ?Response
+    {
+        return null;
+    }
+
+    public function onAuthenticationFailure(Request $request, AuthenticationException $exception): ?Response
+    {
+        if (null !== $this->logger) {
+            $this->logger->info('Basic authentication failed for user.', ['username' => $request->headers->get('PHP_AUTH_USER'), 'exception' => $exception]);
+        }
+
+        return $this->start($request, $exception);
+    }
+
+    /** @todo not sure if true or false */
```

Comment:
IIRC, with basic authentication, the browser session is forgotten as soon as the user closes the browser.

Reply:
Thanks for your review. I just checked the old `BasicAuthenticationListener` and I didn't see any rememberservice being used there, so it indeed seems like HttpBasic never supported remember me.